### PR TITLE
Refactor Dapr Bot

### PR DESCRIPTION
**Pull request by shubham1172**
# Description

Move the GitHub action script into a separate file, refactor code. This will make is easier to maintain and enhance the bot.

## Issue reference

Related: #5326 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .github/scripts/dapr_bot.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
+// list of owner who can control dapr-bot workflow
+// TODO: Read owners from OWNERS file.
```
Comment (ItalyPaleAle) [inline .github/scripts/dapr_bot.js:2]: Since this is JS code, do you think this TODO could actually be made?

**Review by ItalyPaleAle — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

File: .github/scripts/dapr_bot.js (hunk 1 above)
Comment (ItalyPaleAle) [reply .github/scripts/dapr_bot.js:2]: (can be a separate PR :) )

Comment (shubham1172) [reply .github/scripts/dapr_bot.js:2]: Thanks @ItalyPaleAle, created https://github.com/dapr/dapr/issues/5405

**Review by daixiang0 — APPROVED** [review]

**Comment by daixiang0** [comment]
@shubham1172 thanks!

**Pull request merged**